modules/gdscript/gdscript.cpp

```diff
@@ -111,22 +111,22 @@ GDScriptInstance *GDScript::_create_instance(const Variant **p_args, int p_argco
 	GDScriptLanguage::singleton->lock->unlock();
 #endif
 
-	initializer->call(instance, p_args, p_argcount, r_error);
+	if (p_argcount >= 0) {
+		initializer->call(instance, p_args, p_argcount, r_error);
 
-	if (r_error.error != Variant::CallError::CALL_OK) {
-		instance->script = Ref<GDScript>();
-		instance->owner->set_script_instance(NULL);
+		if (r_error.error != Variant::CallError::CALL_OK) {
+			instance->script = Ref<GDScript>();
+			instance->owner->set_script_instance(NULL);
 #ifndef NO_THREADS
-		GDScriptLanguage::singleton->lock->lock();
+			GDScriptLanguage::singleton->lock->lock();
 #endif
-		instances.erase(p_owner);
+			instances.erase(p_owner);
 #ifndef NO_THREADS
-		GDScriptLanguage::singleton->lock->unlock();
+			GDScriptLanguage::singleton->lock->unlock();
 #endif
-
-		ERR_FAIL_COND_V(r_error.error != Variant::CallError::CALL_OK, NULL); //error constructing
+			ERR_FAIL_V_MSG(NULL, "Error constructing a GDScriptInstance.");
```

Comment:
why this always has to fail?

Reply:
Because the error is handled in the above `if` statement, and having to check the same condition within `ERR_FAIL_COND` is redundant (it always fails here regardless).